SportHub/src/modules/AdminMainArticlesSection/components/AddMainArticle/components/AddMainArticle.js

```diff
@@ -0,0 +1,105 @@
+
+    const [subcategory, setSubcategory] = useState(selectedSubcategory)
+    useEffect(() => setSubcategory(selectedSubcategory), [selectedSubcategory])
+
+    const [team, setTeam] = useState(selectedTeam)
+    useEffect(() => setTeam(selectedTeam), [selectedTeam])
+
+    const [article, setArticle] = useState(selectedArticle)
+    useEffect(() => setArticle(selectedArticle), [selectedArticle])
+
+    const [disabled, setDisabled] = useState(category === undefined || category === null)
+
+    useEffect(() => {
+        setDisabled(category === undefined || category === null)
+        setSubcategory(null)
+        setTeam(null)
+        setArticle(null)
+        if(category === selectedCategory){
+            setSubcategory(selectedSubcategory)
+            setTeam(selectedTeam)
+            setArticle(selectedArticle)
+        }
+    }, [category])
+
+    useEffect(() => {
+        SaveData(order, category, subcategory, team, article)
+    }, [order, category, subcategory, team, article])
+
+    const GenerateAddOneMoreArticleButton = () => {
+        if (isLastMainArticle)
+            return(
+            <div className={styles.add_one_more_article} onClick={() => AddNewMainArticle()}>
+                + Add one more article
+            </div>
+        )
+    }
+
+    return (
+        <div>
+            <div className={styles.content}>
+                <div className={styles.upper_autocompletes}>
+                    <AutoComplete
+                        label={"Category*"}
+                        width={"17vw"}
+                        value={selectedCategory}
+                        setValue={setCategory}
+                        options={OPTIONS}
+                        areOptionsObjects={true}
+                        optionLable={"name"}
+                        propertyToCompare={"id"} />
+                    <AutoComplete
+                        label={"Subcategory"}
+                        width={"17vw"}
+                        value={selectedSubcategory}
+                        setValue={setSubcategory}
+                        disabled={disabled}
+                        options={OPTIONS}
+                        areOptionsObjects={true}
+                        optionLable={"name"}
+                        propertyToCompare={"id"} />
+                    <AutoComplete
+                        label={"Team"}
+                        width={"17vw"}
+                        value={selectedTeam}
+                        setValue={setTeam}
+                        disabled={disabled}
+                        options={OPTIONS}
+                        areOptionsObjects={true}
+                        optionLable={"name"}
+                        propertyToCompare={"id"} />
+                </div>
+                <div className={styles.add_article_autocomplete}>
+                    <AutoComplete
+                        label={"Article*"}
+                        width={"100%"}
```

Comment:
don't like the idea of passing width as prop. Consider using containers with width that wrap autoCompletes. (those three with display flex and this one will be 100%)